modules/backendscheduler/work/job.go

```diff
@@ -120,13 +120,19 @@ func (j *Job) GetWorkerID() string {
 func (j *Job) OnBlock(id string) bool {
 	for _, b := range j.GetCompactionInput() {
 		if b == id {
-			return true
+			switch j.GetStatus() {
+			case tempopb.JobStatus_JOB_STATUS_RUNNING, tempopb.JobStatus_JOB_STATUS_UNSPECIFIED:
```

Comment:
In low block throughput environments, it was enough for any status.  Observations in a higher throughput environment mean that the output blocks of level 0 don't get fed into the next compaction level 1, etc jobs, resulting in lower compaction throughput.

I believe we will only care about the running and unspecified when deciding work for future jobs as well.